proxy/src/main/java/com/velocitypowered/proxy/VelocityServer.java

```diff
@@ -332,18 +332,41 @@ private void loadPlugins() {
 
     try {
       Path pluginPath = Path.of("plugins");
+      Path updatePath = pluginPath.resolve("to-update");
+      Path outdatedPluginsPath = pluginPath.resolve("outdated");
 
-      if (!pluginPath.toFile().exists()) {
+      if (Files.notExists(pluginPath)) {
         Files.createDirectory(pluginPath);
-      } else {
-        if (!pluginPath.toFile().isDirectory()) {
-          logger.warn("Plugin location {} is not a directory, continuing without loading plugins",
-              pluginPath);
-          return;
-        }
+      }
+      if (Files.notExists(updatePath)) {
+        Files.createDirectory(updatePath);
+      }
+      if (Files.notExists(outdatedPluginsPath)) {
+        Files.createDirectory(outdatedPluginsPath);
+      }
```

Comment:
The creation of plugin update folders could be moved inside the [`VelocityConfiguration#isAutoUpdate`](https://github.com/PaperMC/Velocity/pull/842/files#diff-1217e64751ab6522195a858620cceeb4f2d2b6506c36dd4e396726ad3e7ef0cbR354) check, to avoid creating these folders unnecessarily in case someone doesn't want this function

Reply:
Should be resolved